From 80b7a4276abb832b6af1d1d060affb0988ab7fa0 Mon Sep 17 00:00:00 2001
From: jiangchuangang <jiangchuangang@huawei.com>
Date: Tue, 30 Nov 2021 15:25:27 +0800
Subject: [PATCH] fix ConditionPathIsReadWrite when path does not exist.

(cherry picked from commit 7c4c9948d02ceda903ed4e4deea0d0084612625a)

Conflict:NA
Reference:https://github.com/systemd/systemd/commit/80b7a4276abb832b6af1d1d060affb0988ab7fa0
---
 src/shared/condition.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/shared/condition.c b/src/shared/condition.c
index 163ddc1208..6645f771dd 100644
--- a/src/shared/condition.c
+++ b/src/shared/condition.c
@@ -897,11 +897,15 @@ static int condition_test_path_is_mount_point(Condition *c, char **env) {
 }
 
 static int condition_test_path_is_read_write(Condition *c, char **env) {
+        int r;
+
         assert(c);
         assert(c->parameter);
         assert(c->type == CONDITION_PATH_IS_READ_WRITE);
 
-        return path_is_read_only_fs(c->parameter) <= 0;
+        r = path_is_read_only_fs(c->parameter);
+
+        return r <= 0 && r != -ENOENT;
 }
 
 static int condition_test_cpufeature(Condition *c, char **env) {
-- 
2.33.0

